r2dbc-spi/src/main/java/io/r2dbc/spi/RowMetadata.java

```diff
@@ -41,4 +42,14 @@
      */
     Iterable<? extends ColumnMetadata> getColumnMetadatas();
 
+    /**
+     * Returns an unmodifiable collection of unique column names, and any attempts to modify the returned
+     * collection, whether direct or via its iterator, result in an {@link UnsupportedOperationException}.
+     * The order of the column names depends on the actual query and also column name lookup depends on the database
+     * sorting rules and some databases support escape characters to enforce a particular mode of comparison.
+     *
+     * @return the column names.
+     */
+    Collection<String> getColumnNames();
```

Comment:
This new method and `io.r2dbc.spi.ColumnMetadata.getName()` should explicitly describe whether they return real column names or their aliases in Javadoc, because JDBC has different methods for these purposes.

See also more detailed description in
https://github.com/r2dbc/r2dbc-h2/issues/73#issuecomment-487535441

Reply:
Thanks, @Katzyn for bringing this up, I have updated the javadoc.